### ISSUE TITLE
Overview of Tools supporting EE Architecture Design

### ISSUE THREAD

**Issue by marcusvw** [localmotors/veh-engineering#7]
Evaluating  opens source  and comercial solutions supporting one or more of the following steps/properties in System Architecture:

- Requirements Engineering
- Change Management
- File Versioning (SVN/Git like etc.) 
- Logical Architecture
- Software Architecture
- Hardware Architecture / Wire Harness Schematics
- Physical Wire harness Design 
- Test/Validation Management
- Functional Safety (HARA/ FMEA/etc.)
- Customized Agile/SCRUM Development Cycles

Would be a good step to find a fitting and cost effective solution.

**Comment by creighto** [comment]
Absolutely, and we will when we are hitting obstacles that we cannot work-around with what we have right now. 

My wild suggestion is, as we are a really small team, to try and learn all about GitHub and how we could be using that for the first 5 (yes five!) and last 3 bullets.

So a whopping 8 out of 10 tools with a strict "content-oriented" approach, just the text and images for now, using good-old file-folder structure where appropriate. ;-)


"Why leave out the Hardware and Wire Harness bullets, then?" I hear you ask.  Because I don't know enough about which kinds of tools are out there for those. I have only seen how we used KiCAD for the Olli and that might not be what you had in mind… So here's a question: Can you see a way to do those two things also in a gitflow-style environment? Ideally with a git-back-end?

**Comment by creighto** [comment]
I took a sneak peek at what you're working on in the according branch. I'm not a big fan of binary files in git. You lose all the good stuff that gitflow brings.

And content-wise, I think we are simply not yet ready to start with a tool evaluation. I'd like to move this ticket on the Kanban board all the way down, if you follow what I'm saying.

First we will need agreement on *what it really is* that we are working on, e.g. a piece of wood, before we evaluate all kinds of wood-working tools. Again, I'd like to follow a whole-heartedly pragmatic approach of "content first." Let's try to roughly structure our requirements and system architecture using folders and text+images files (cf #6 ) before evaluating different representations and tools. If we keep the proposed workflow of propose-discuss-merge, we have full traceability of our decisions.

Can you agree to my suggestion?

**Comment by marcusvw** [comment]
I started from my point of experience, we used PreeVision for overall Vehicle Architecture. So my idea was just starting to collect solutions which are used for the tasks that we have done there. The overall Vehicle Architecture Functionality/Requirements is the Starting point for the safety process/ Item definition and Hazard and Risk Analysis.

A general point regarding the tool chain is that, what we are actually not doing now, have to be done with additional effort later. 
Because you need a complete traceability regarding ISO26262, you can do this probably manual, but actually I'm not sure if you need a metric over the traceability (referencing requirements with artifacts from the Logical Architecture and the SW Architecture, etc.). In general doing the manual, it means that you have to reference each item that is related to an safety artifact must be referenced by a unique id and linked via a spreadsheet etc. to for example the requirements. Doing this without tool support will be difficult. 

We will need from my point of view, for the Logical Architecture and the Software Architecture a graphical "expression". A lot of things getting to complex only describing it by text. There are several possibilities like UML etc. 
Something between graphical and textual would be AADL with the supported tools, but this is mainly a "transport" format.

We should first check neither if it is possible to have a manually realized traceability to full fill iso26262 or if we need here something with metrics. From my point of view we will need here metrics, see also: http://www.kpit.com/resources/downloads/medini-analyze/16-prsentation-ikv-matlab.pdf (unluckily in German)

For me the spreadsheet is actually only a collection of ideas without any evaluation. For sure we have to clarify more what will be "the job". Actually I've excluded ECU (Hardware) Development. Only Wire harness Schematic and also the physical wire harness is the "physical part". For a stringent development we need also reference each wire etc. that is related to safety requirements, so it make sense also to cover this in a tool chain.   
From my point of view PreeVision might be a quite expensive solution, (its actually the industry leading solution for such a job) I searched for alternatives. 

We should have the possibility to use spread sheet. Especially for comparisons etc.
